paas_app_charmer/observability.py

```diff
@@ -47,13 +48,24 @@ def __init__(  # pylint: disable=too-many-arguments
             jobs=jobs,
             relation_name="metrics-endpoint",
         )
-        self._logging = LogProxyConsumer(
-            charm,
-            alert_rules_path=os.path.join(cos_dir, "loki_alert_rules"),
-            container_name=container_name,
-            log_files=[str(log_file) for log_file in log_files],
-            relation_name="logging",
-        )
+        if enable_pebble_log_forwarding():
+            # ignore "import outside toplevel" linting error
+            import charms.loki_k8s.v1.loki_push_api  # pylint: disable=import-outside-toplevel
+
+            self._logging = charms.loki_k8s.v1.loki_push_api.LogForwarder(
```

Comment:
Out of curiosity: What happens if you deploy them charm in a Juju version then the charm uses the old library and then you upgrade Juju? Is there any action that should be taken?

Reply:
Yes, that's a good question. If the charm is only bundled with the old version of the library, it will never use the Pebble log forwarding feature.